### PR TITLE
fix(core): fix partition directory left on disk when latest dropped

### PR DESCRIPTION
**Pull request by ideoma**
Now that QuestDB allows to drop last (active) partition the Async `O3PurgePartitionJob` need to be adjusted to the use case and remove the directory from disk.

Also here added more logging on what `O3PurgePartitionJob` tries to delete and return error when partition DROP or DETACH with WHERE clause do not find suitable partition to detach to have same feedback as dropping with LISTing partition syntax.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 34 (94.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|17|18|94.44%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 34 (94.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|17|18|94.44%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|

File: core/src/test/java/io/questdb/griffin/O3PartitionPurgeTest.java [hunk 1]
```diff
@@ -356,6 +356,71 @@ public void testPartitionDeletedAsyncAfterDroppedBySql() throws Exception {
         });
     }
 
+    @Test
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/griffin/O3PartitionPurgeTest.java:359]: file doesn't seem to be formatted correctly

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/test/java/io/questdb/griffin/O3PartitionPurgeTest.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/O3PartitionPurgeTest.java]: fixed in https://github.com/questdb/questdb/pull/2802/commits/e8545c9e00a159392d3674a6c27f8840075fb742

**Review by bluestreak01 — APPROVED** [review]

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 44 / 46 (95.65%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/log/SyncLogger.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|17|18|94.44%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByAllIndexedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/DatabaseSnapshotAgent.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableStructureAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/model/InsertModel.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/GenericRecordMetadata.java|3|3|100.00%|

**Pull request merged**